src/devices/bus/pce_ctrl/pachinko.cpp

```diff
@@ -0,0 +1,163 @@
+    Pachio-kun games. However, only Maboroshi no Densetsu and Juuban Shoubu
+    prompt for controller type at boot. How to enable on the other two games?
+
+**********************************************************************/
+
+#include "emu.h"
+#include "pachinko.h"
+
+
+
+//**************************************************************************
+//  DEVICE DEFINITIONS
+//**************************************************************************
+
+DEFINE_DEVICE_TYPE(PCE_PACHINKO, pce_pachinko_device, "pce_pachinko", "Coconuts Japan Pachinko Controller CJPC-101")
+
+
+static INPUT_PORTS_START( pce_pachinko )
+	// II is left of I on the original pad so we map them in reverse order
+	PORT_START("BUTTONS")
+	PORT_BIT( 0x01, IP_ACTIVE_LOW, IPT_BUTTON2 ) PORT_NAME("%p Button I")
+	PORT_BIT( 0x02, IP_ACTIVE_LOW, IPT_BUTTON1 ) PORT_NAME("%p Button II")
+	PORT_BIT( 0x04, IP_ACTIVE_LOW, IPT_SELECT  ) PORT_NAME("%p Select")
+	PORT_BIT( 0x08, IP_ACTIVE_LOW, IPT_START   ) PORT_NAME("%p Run")
+
+	PORT_START("DPAD")
+	PORT_BIT( 0x01, IP_ACTIVE_LOW, IPT_JOYSTICK_UP )    PORT_8WAY
+	PORT_BIT( 0x02, IP_ACTIVE_LOW, IPT_JOYSTICK_RIGHT ) PORT_8WAY
+	PORT_BIT( 0x04, IP_ACTIVE_LOW, IPT_JOYSTICK_DOWN )  PORT_8WAY
+	PORT_BIT( 0x08, IP_ACTIVE_LOW, IPT_JOYSTICK_LEFT )  PORT_8WAY
+
+	PORT_START("TRIGGER")
+	PORT_BIT( 0xff, 0x80, IPT_PEDAL ) PORT_MINMAX(0x80, 0xff) PORT_SENSITIVITY(10) PORT_KEYDELTA(20) PORT_INVERT PORT_REVERSE
+INPUT_PORTS_END
+
+
+//-------------------------------------------------
+//  input_ports - device-specific input ports
+//-------------------------------------------------
+
+ioport_constructor pce_pachinko_device::device_input_ports() const
+{
+	return INPUT_PORTS_NAME( pce_pachinko );
+}
+
+
+
+//**************************************************************************
+//  LIVE DEVICE
+//**************************************************************************
+
+//-------------------------------------------------
+//  pce_pachinko_device - constructor
+//-------------------------------------------------
+
+pce_pachinko_device::pce_pachinko_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock) :
+	device_t(mconfig, PCE_PACHINKO, tag, owner, clock),
+	device_pce_control_port_interface(mconfig, *this),
+	m_buttons(*this, "BUTTONS"),
+	m_dpad(*this, "DPAD"),
+	m_trigger(*this, "TRIGGER"),
+	m_counter(0),
+	m_prev_sel(false),
+	m_prev_clr(false)
+{
+}
+
+
+//-------------------------------------------------
+//  device_start - device-specific startup
+//-------------------------------------------------
+
+void pce_pachinko_device::device_start()
+{
+	save_item(NAME(m_counter));
+	save_item(NAME(m_prev_sel));
+	save_item(NAME(m_prev_clr));
+}
+
+
+//-------------------------------------------------
+//  device_reset - device-specific reset
+//-------------------------------------------------
+
+void pce_pachinko_device::device_reset()
+{
+	m_counter = 0;
+	m_prev_sel = false;
+	m_prev_clr = false;
+}
```

Comment:
Does system reset actually propagate to the controllers?  If it doesn’t, they shouldn’t respond to reset.

Also, `m_prev_sel` and `m_prev_clr` are input states, so making them magically change like this can cause trouble as it can get out-of-sync with what the device on the other side of the connection is doing.  If the thing on the other side of the connection changes its outputs on reset, it will push them through the usual callbacks.

Reply:
Removed, although I suppose hypothetically it could pull the CLR pin high or something on reset. The only signals out to the controller are the SEL and CLR pins.